tests/e2e/utils/auth.py

```diff
@@ -0,0 +1,13 @@
+from urllib.parse import urlparse
+
+import boto3
+from aws_requests_auth.boto_utils import BotoAWSRequestsAuth
+
+
+def iam_auth(url: str, aws_service: str) -> BotoAWSRequestsAuth:
```

Comment:
```suggestion
def iam_auth(url: str, aws_service: str) -> BotoAWSRequestsAuth:
```

Maybe something more explicit like: `build_iam_auth`, or `sign_iam_auth`, etc.